lib/chef/exceptions.rb

```diff
@@ -97,6 +97,7 @@ class GroupIDNotFound < ArgumentError; end
     class ConflictingMembersInGroup < ArgumentError; end
     class InvalidResourceReference < RuntimeError; end
     class ResourceNotFound < RuntimeError; end
+    class ProviderNotFound < ArgumentError; end
```

Comment:
Let's just have this throw NoProviderAvailable, screw ArgumentError, and create an alias for either `NoResourceAvailable` or `ProviderNotFound`.  Got a preference for which?